cmd/launcher/launcher.go

```diff
@@ -194,6 +194,13 @@ func runLauncher(ctx context.Context, cancel func(), multiSlogger, systemMultiSl
 	flagController := flags.NewFlagController(slogger, stores[storage.AgentFlagsStore], fcOpts...)
 	k := knapsack.New(stores, flagController, db, multiSlogger, systemMultiSlogger)
 
+	// Generate a new run ID
+	newRunID := k.GetRunID()
```

Comment:
I know this has merged, and it's fine, but I'd probably skip the variable here, and put `k.GetRunID()` into the two `With` functions